### PR TITLE
separate CI config for master & PRs

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
Our CI configuration is becoming unwieldy; this PRs attempt to get back some measure of control by splitting the CI configuration of the master branch and the CI configuration of PRs, as there are a number of jobs that should run only on one of them.

Because there is a bit of setup involved (at the Azure level), this PR does not attempt to remove all duplication; that will be done as future PRs.

I have kept the changes in separate commits for ease of reviewing: the first commit just copies the entire `azure-piplines.yml` file to `ci/prs.yml`, and the second commit prunes from each what they should never run. Reviewers should focus on the second commit.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -40,73 +32,8 @@ jobs:
...
-              exit 1
-          fi
-
-  - template: ci/da-ghc-lib/compile.yml
-    parameters:
-      final_job_name: da_ghc_lib
-
   - job: Linux
     dependsOn:
-      - da_ghc_lib
```
Comment (cocreature) [inline azure-pipelines.yml:109]: Why do we no longer depend on this? Same question below.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -318,10 +234,8 @@ jobs:
 
   - job: compatibility_windows
     dependsOn:
-      - da_ghc_lib
       - check_for_release
       - compatibility_ts_libs
-      - patch_bazel_windows
```
Comment (cocreature) [inline azure-pipelines.yml:324]: And same question for this.

File: ci/prs.yml [hunk 3]
```diff
@@ -0,0 +1,612 @@
...
+      demands: assignment -equals default
+    steps:
+      - bash: |
+          set -euo pipefail
+          if [ "$(Build.Reason)" == "PullRequest" ]; then
+              echo "##vso[task.setvariable variable=branch;isOutput=true]$(git rev-parse HEAD^2)"
+              echo "##vso[task.setvariable variable=master;isOutput=true]$(git rev-parse HEAD^1)"
+              echo "##vso[task.setvariable variable=fork_point;isOutput=true]$(git merge-base $(git rev-parse HEAD^1) $(git rev-parse HEAD^2))"
+          else
+              echo "##vso[task.setvariable variable=branch;isOutput=true]$(git rev-parse HEAD)"
```
Comment (cocreature) [inline ci/prs.yml:27]: When do we expect to hit this? Manually queued builds? Shouldn’t those still be a branch forked off from master?

File: ci/prs.yml [hunk 4]
```diff
@@ -0,0 +1,612 @@
...
+      - bash: ci/dev-env-install.sh
+        displayName: 'Build/Install the Developer Environment'
+      - bash: |
+          source dev-env/lib/ensure-nix
+          ci/dev-env-push.py
+        displayName: 'Push Developer Environment build results'
+        condition: eq(variables['System.PullRequest.IsFork'], 'False')
+        env:
+          # to upload to the Nix cache
+          GOOGLE_APPLICATION_CREDENTIALS_CONTENT: $(GOOGLE_APPLICATION_CREDENTIALS_CONTENT)
```
Comment (cocreature) [inline ci/prs.yml:236]: I thought we don’t upload on PRs?

**Review by cocreature — APPROVED** [review]
LGTM, I have to admit that just from looking at this PR this seems even messier but open to seeing if you can factor out enough of the shared logic to make it cleaner than the current state.

File: azure-pipelines.yml (hunk 1 above)
Comment (garyverhaegen-da) [reply azure-pipelines.yml:109]: Open to discussion on this; my reasoning here was that master builds should not try to build those.

Comment (cocreature) [reply azure-pipelines.yml:109]: Do we have cache expiration on those? If so, I’d definitely prefer to keep this in. If not, I guess, it’s probably safe to not build this but it also seems nice to have this be self-contained and not rely on the fact that PRs built this before.

File: ci/prs.yml [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+      - bash: ci/configure-bazel.sh
+        displayName: 'Configure Bazel'
+        env:
+          IS_FORK: $(System.PullRequest.IsFork)
+          # to upload to the bazel cache
+          GOOGLE_APPLICATION_CREDENTIALS_CONTENT: $(GOOGLE_APPLICATION_CREDENTIALS_CONTENT)
+      - bash: |
+          set -euo pipefail
+          eval "$(./dev-env/bin/dade-assist)"
+          bazel test --config scala_2_13 //libs-scala/ports/...
```
Comment (cocreature) [inline ci/prs.yml:242]: this has just changed on master

**Comment by garyverhaegen-da** [comment]
Superseded by #8374.

**Pull request closed without merging**